pkg/controller/manager/sync.go

```diff
@@ -108,6 +108,10 @@ func (c *Controller) sync(ctx context.Context, key string) error {
 		return nil
 	}
 
+	if helpers.IsStatusConditionPresentAndTrue(sc.Status.Conditions, scyllav1.ProgressingCondition, sc.Generation) {
```

Comment:
agent token is shared between all nodes in cluster, you don't have to wait until entire cluster is deployed. It's enough to wait for auth token progressing condition == false